test/accounts.dart

```diff
@@ -0,0 +1,376 @@
+
+        final completer = Completer<bool>();
+        String txHash = "";
+        String updateTexHash = "";
+
+        kc2Service.updateUserCallback = (tx) async {
+          debugPrint('>> update user update 1 called');
+          if (tx.failedReason != null) {
+            completer.complete(false);
+            return;
+          }
+
+          if (tx.hash != updateTexHash) {
+            expect(tx.hash, updateTexHash);
+            completer.complete(false);
+            return;
+          }
+
+          expect(tx.phoneNumberHash, newPhoneNumberHash);
+          expect(tx.username, katyaNewPhoneNumber);
+          expect(tx.signer, katya.accountId);
+
+          // all 3 methods should return's Katya's account data
+          KC2UserInfo? userInfo =
+              await kc2Service.getUserInfoByAccountId(katya.accountId);
+
+          if (userInfo == null) {
+            debugPrint('Faied to get user info by account id');
+            completer.complete(false);
+            return;
+          }
+
+          expect(userInfo.accountId, katya.accountId);
+          expect(userInfo.phoneNumberHash, '0x$newPhoneNumberHash');
+          expect(userInfo.userName, katyaUserName);
+
+          // get the user by updated phone number
+          userInfo =
+              await kc2Service.getUserInfoByPhoneNumberHash(newPhoneNumberHash);
+
+          if (userInfo == null) {
+            debugPrint('Faied to get user info by phone number');
+            completer.complete(false);
+            return;
+          }
+
+          expect(userInfo.accountId, katya.accountId);
+          expect(userInfo.phoneNumberHash, '0x$newPhoneNumberHash');
+          expect(userInfo.userName, katyaUserName);
+
+          userInfo = await kc2Service.getUserInfoByUserName(katyaUserName);
+          if (userInfo == null) {
+            debugPrint('Faied to get user info by nickname');
+            completer.complete(false);
+            return;
+          }
+
+          expect(userInfo.accountId, katya.accountId);
+          expect(userInfo.phoneNumberHash, '0x$newPhoneNumberHash');
+          expect(userInfo.userName, katyaUserName);
+
+          completer.complete(true);
+        };
+
+        kc2Service.newUserCallback = (tx) async {
+          debugPrint('>> new user callback called');
+          if (tx.failedReason != null) {
+            completer.complete(false);
+            return;
+          }
+
+          if (tx.hash != txHash) {
+            debugPrint('unexecpted tx hash: ${tx.hash} ');
+            completer.complete(false);
+            return;
+          }
+
+          // @Danylo Kyrieiev: this tx submission fails
+          debugPrint('calling update user...');
+          try {
+            updateTexHash =
+                await kc2Service.updateUser(katyaUserName, katyaNewPhoneNumber);
+          } catch (e) {
+            debugPrint('Failed to update user: $e');
+            completer.complete(false);
+          }
+        };
```

Comment:
Here we have to mistakes. First you try to pass phone number and not phone number hash. Second is that you pass username that don't changed so transaction tries to update username to same and that lead to an error. I will fix it in nest commit